src/lib/operators/aggregate_sort.cpp

```diff
@@ -368,14 +427,41 @@ std::shared_ptr<const Table> AggregateSort::_on_execute() {
    * However, we did not benchmark it, so we cannot prove it.
    */
 
-  // Sort input table consecutively by the group by columns (stable sort)
-  auto sorted_table = input_table;
-  for (const auto& column_id : _groupby_column_ids) {
-    const auto sorted_wrapper = std::make_shared<TableWrapper>(sorted_table);
-    sorted_wrapper->execute();
-    Sort sort = Sort(sorted_wrapper, column_id);
-    sort.execute();
-    sorted_table = sort.get_output();
+  auto table_value_clustered_by = input_table->value_clustered_by();
+
+  std::shared_ptr<const Table> sorted_table = input_table;
+  if (!_groupby_column_ids.empty()) {
+    /**
+     * Check if the value clustering matches the columns we are grouping by
+     * (these are the columns we would need to sort by to achieve value clustering).
+     * If the input is already value clustered according to our needs, this means that values that would end up
+     * in the same cluster are already in the same chunk.
+     * Thus, we can execute all remaining sorts on a per-chunks basis.
+     * If the value clustering doesn't match the columns we group by, we need to re-sort the whole table
+     * to achieve the value clustering we need.
+     */
+    if (table_value_clustered_by && !table_value_clustered_by->empty() &&
```

Comment:
You are right. Because value_clustering on tables can not be removed, it is indeed no valid state and a check for `table_value_clustered_by` should be sufficient